# P3_Réalisez une étude de santé publique

Mise en situation
Vous êtes intégré à une nouvelle équipe de chercheurs de la Food and Agriculture Organization of the United Nations (FAO), l'un des organes qui compose l'ONU et dont l'objectif est d' « aider à construire un monde libéré de la faim ».

Votre équipe est chargée de réaliser une étude de grande ampleur sur le thème de la sous-nutrition dans le monde.

Le problème de la faim est complexe et peut avoir de multiples causes, différentes selon les pays. L’étape préliminaire de cette étude sera donc d’établir un “état de l’art” des recherches déjà publiées, mais également de mener une étude statistique destinée à orienter les recherches vers des pays particuliers, et de mettre en lumière différentes causes de la faim. Ainsi, une poignée de data analysts (dont vous !) a été sélectionnée pour mener cette étape préliminaire. Lors de la première réunion, vous avez été désigné pour mettre une place la base de données que votre équipe pourra requéter (en SQL) à souhait pour réaliser cette étude statistique.

 

Les données
Les données sont disponibles sur ce lien et sont constituées de 5 fichiers:

fr_animaux.csv : multiples indicateurs de production des produits animaux en 2013
fr_population.csv: population mondiale par pays en 2013
fr_vegetaux.csv: multiples indicateurs de production des produits végétaux en 2013
fr_céréales.csv: quantité de céréales produites au niveau mondial en 2013    
fr_sousalimentation.csv: nombre de personnes sous alimentées dans le monde de 2013 à 2017.

## Importer  des librairies Python et lire des bases de données

In [72]:
#impotter des librairies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [73]:
#importer des database
df_population = pd.read_csv('fr_population.csv')
df_animaux = pd.read_csv('fr_animaux.csv')
df_vegetaux = pd.read_csv('fr_vegetaux.csv')
df_céréales = pd.read_csv('fr_céréales.csv')
df_sousalimentation = pd.read_csv('fr_sousalimentation.csv')

## Connaître la base de données
df_population 
df_animaux 
df_vegetaux 
df_céréales 
df_sousalimentation 

In [74]:
df_population.head()

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        202   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          3   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          4   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...         79   

             Zone  Code Élément            Élément  Code Produit     Produit  \
0     Afghanistan           511  Population totale          2501  Population   
1  Afrique du Sud           511  Population totale          2501  Population   
2         Albanie           511  Population totale          2501  Population   
3         Algérie           511  Population totale          2501  Population   
4       Allemagne           511  Population totale          2501  Population   

   Code année  Année           Unité  Valeur Symbole Description du Symbole  
0        2013   2013  1000 personnes   30552     NaN      Donnée officielle  
1        2013   2013  1000 personnes   52776     NaN      Donnée officielle  
2        2013   2013  1000 personnes    3173     NaN      Donnée officielle  
3        2013   2013  1000 personnes   39208     NaN      Donnée officielle  
4        2013   2013  1000 personnes   82727     NaN      Donnée officielle

In [75]:
df_vegetaux.sample(8)

Code Domaine                                            Domaine  \
93007          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
52447          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
57808          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
81121          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
18771          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
93744          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
101179         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
11794          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

        Code zone                       Zone  Code Élément  \
93007         210                      Suède          5123   
52447         110                      Japon          5511   
57808         123                    Libéria           664   
81121          37  République centrafricaine           645   
18771          33                     Canada          5142   
93744         211                     Suisse           645   
101179        234                    Uruguay           664   
11794          53                      Bénin          5131   

                                                  Élément  Code Produit  \
93007                                              Pertes          2515   
52447                                          Production          2605   
57808      Disponibilité alimentaire (Kcal/personne/jour)          2549   
81121   Disponibilité alimentaire en quantité (kg/pers...          2602   
18771                                          Nourriture          2531   
93744   Disponibilité alimentaire en quantité (kg/pers...          2531   
101179     Disponibilité alimentaire (Kcal/personne/jour)          2611   
11794                                          Traitement          2559   

                    Produit  Code année  Année               Unité   Valeur  \
93007                Seigle        2013   2013  Milliers de tonnes     3.00   
52447       Légumes, Autres        2013   2013  Milliers de tonnes  9461.00   
57808   Légumineuses Autres        2013   2013  Kcal/personne/jour    13.00   
81121               Oignons        2013   2013                  kg     0.00   
18771       Pommes de Terre        2013   2013  Milliers de tonnes  2501.00   
93744       Pommes de Terre        2013   2013                  kg    40.73   
101179  Oranges, Mandarines        2013   2013  Kcal/personne/jour    16.00   
11794      Graines de coton        2013   2013  Milliers de tonnes    90.00   

       Symbole Description du Symbole  
93007        S  Données standardisées  
52447        S  Données standardisées  
57808       Fc        Donnée calculée  
81121       Fc        Donnée calculée  
18771        S  Données standardisées  
93744       Fc        Donnée calculée  
101179      Fc        Donnée calculée  
11794        S  Données standardisées

In [76]:
df_population.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole'],
      dtype='object')

Reconnaître, vérifier et nettoyer les données

In [77]:
df_population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Code Domaine            175 non-null    object
 1   Domaine                 175 non-null    object
 2   Code zone               175 non-null    int64 
 3   Zone                    175 non-null    object
 4   Code Élément            175 non-null    int64 
 5   Élément                 175 non-null    object
 6   Code Produit            175 non-null    int64 
 7   Produit                 175 non-null    object
 8   Code année              175 non-null    int64 
 9   Année                   175 non-null    int64 
 10  Unité                   175 non-null    object
 11  Valeur                  175 non-null    int64 
 12  Symbole                 1 non-null      object
 13  Description du Symbole  175 non-null    object
dtypes: int64(6), object(8)
memory usage: 19.3+ KB


In [78]:
df_animaux.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37166 entries, 0 to 37165
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            37166 non-null  object 
 1   Domaine                 37166 non-null  object 
 2   Code zone               37166 non-null  int64  
 3   Zone                    37166 non-null  object 
 4   Code Élément            37166 non-null  int64  
 5   Élément                 37166 non-null  object 
 6   Code Produit            37166 non-null  int64  
 7   Produit                 37166 non-null  object 
 8   Code année              37166 non-null  int64  
 9   Année                   37166 non-null  int64  
 10  Unité                   37166 non-null  object 
 11  Valeur                  37166 non-null  float64
 12  Symbole                 37166 non-null  object 
 13  Description du Symbole  37166 non-null  object 
dtypes: float64(1), int64(5), object(8)
mem

In [79]:
df_céréales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Code Domaine            891 non-null    object
 1   Domaine                 891 non-null    object
 2   Code zone               891 non-null    int64 
 3   Zone                    891 non-null    object
 4   Code Élément            891 non-null    int64 
 5   Élément                 891 non-null    object
 6   Code Produit            891 non-null    int64 
 7   Produit                 891 non-null    object
 8   Code année              891 non-null    int64 
 9   Année                   891 non-null    int64 
 10  Unité                   891 non-null    object
 11  Valeur                  891 non-null    int64 
 12  Symbole                 891 non-null    object
 13  Description du Symbole  891 non-null    object
dtypes: int64(6), object(8)
memory usage: 97.6+ KB


In [80]:
df_sousalimentation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            1020 non-null   object 
 1   Domaine                 1020 non-null   object 
 2   Code zone               1020 non-null   int64  
 3   Zone                    1020 non-null   object 
 4   Code Élément            1020 non-null   int64  
 5   Élément                 1020 non-null   object 
 6   Code Produit            1020 non-null   int64  
 7   Produit                 1020 non-null   object 
 8   Code année              1020 non-null   int64  
 9   Année                   1020 non-null   object 
 10  Unité                   1020 non-null   object 
 11  Valeur                  605 non-null    object 
 12  Symbole                 1020 non-null   object 
 13  Description du Symbole  1020 non-null   object 
 14  Note                    0 non-null      

In [81]:
df_sousalimentation.replace(np.nan, 0, inplace=True)

In [82]:
df_sousalimentation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            1020 non-null   object 
 1   Domaine                 1020 non-null   object 
 2   Code zone               1020 non-null   int64  
 3   Zone                    1020 non-null   object 
 4   Code Élément            1020 non-null   int64  
 5   Élément                 1020 non-null   object 
 6   Code Produit            1020 non-null   int64  
 7   Produit                 1020 non-null   object 
 8   Code année              1020 non-null   int64  
 9   Année                   1020 non-null   object 
 10  Unité                   1020 non-null   object 
 11  Valeur                  1020 non-null   object 
 12  Symbole                 1020 non-null   object 
 13  Description du Symbole  1020 non-null   object 
 14  Note                    1020 non-null   

In [83]:
df_vegetaux.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104871 entries, 0 to 104870
Data columns (total 14 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Code Domaine            104871 non-null  object 
 1   Domaine                 104871 non-null  object 
 2   Code zone               104871 non-null  int64  
 3   Zone                    104871 non-null  object 
 4   Code Élément            104871 non-null  int64  
 5   Élément                 104871 non-null  object 
 6   Code Produit            104871 non-null  int64  
 7   Produit                 104871 non-null  object 
 8   Code année              104871 non-null  int64  
 9   Année                   104871 non-null  int64  
 10  Unité                   104871 non-null  object 
 11  Valeur                  104871 non-null  float64
 12  Symbole                 104871 non-null  object 
 13  Description du Symbole  104871 non-null  object 
dtypes: float64(1), int64

# Question 1 : donnez le résultat de votre calcul pour l'année 2013.
Créez un dataframe contenant les informations de population de chaque pays. Calculez le nombre total d’humains sur la planète. Critiquez votre résultat. En cas d’anomalie, analysez et effectuer les corrections nécessaires.


In [84]:
#Assurez que les données ne concernent que 2013.
df_population['Année'] .nunique()

1

Il n'y a que l'année 2013 dans la colonne année.
On peut additionner la population de chaque pays pour obtenir la population totale du monde

In [85]:
#somme la population des pays pour obtenir la population mondiale
population_2013 = (df_population['Valeur'].sum()*1000)
population_2013

8413993000

In [86]:
print('La population de la terre est' ,population_2013, 'Mais en 2013, la population de la Terre, selon les  Nations Unies, était de 7 210 581 976 (https://www.worldometers.info/world-population/world-population-by-year/)                                  ')

La population de la terre est 8413993000 Mais en 2013, la population de la Terre, selon les  Nations Unies, était de 7 210 581 976 (https://www.worldometers.info/world-population/world-population-by-year/)                                  


In [87]:
df_population.nunique()

Code Domaine                1
Domaine                     1
Code zone                 175
Zone                      175
Code Élément                1
Élément                     1
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       1
Valeur                    175
Symbole                     1
Description du Symbole      2
dtype: int64

Description du Symbole a deux valeurs, voyons ce qu'elles sont :

In [88]:
df_population ['Description du Symbole']. unique ()

array(['Donnée officielle',
       'Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées'],
      dtype=object)

In [89]:
df_population_pas_officielles= (df_population['Description du Symbole']=='Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées')
df_population[df_population_pas_officielles]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

La Chine est comptée deux fois, dans la ligne 33 les données non officielles sont en double
Si nous supprimons) la ligne contenant l'index 33 de notre ensemble de données.

In [90]:
df_population[df_population_pas_officielles]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [91]:
nb_population= df_population['Valeur'].sum()*1000 - 1416667*1000
nb_population

6997326000

In [92]:
#Supprimer la ligne Chine
df_population = df_population[df_population['Code zone'] != 351]

In [93]:
df_population.loc[30:40, ['Zone','Code zone','Valeur']]

Zone  Code zone   Valeur
30                   Cameroun         32    22254
31                     Canada         33    35182
32                      Chili         40    17620
34   Chine - RAS de Hong-Kong         96     7204
35       Chine - RAS de Macao        128      566
36        Chine, continentale         41  1385567
37  Chine, Taiwan Province de        214    23330
38                     Chypre         50     1141
39                   Colombie         44    48321
40                      Congo         46     4448

In [94]:
df_animaux = df_animaux[df_animaux['Code zone'] != 351]

In [95]:
df_vegetaux = df_vegetaux[df_vegetaux['Code zone'] != 351]

In [96]:
df_céréales = df_céréales[df_céréales['Code zone'] != 351]

In [97]:
df_sousalimentation = df_sousalimentation[df_sousalimentation['Code zone'] != 351 ]

In [98]:
df_population.shape

(174, 14)

In [99]:
df_population ['Zone']. nunique ()

174

Note :

In [100]:
print('La population mondiale est de ',nb_population, '. La différence entre cela et le nombre dans les statistiques des Nations Unies car il y a beaucoup de petits pays non enregistrés dans les données.')

La population mondiale est de  6997326000 . La différence entre cela et le nombre dans les statistiques des Nations Unies car il y a beaucoup de petits pays non enregistrés dans les données.


Il y a 206 pays dans le monde. Parmi eux, 193 sont membres des Nations Unies selon https://fr.wikipedia.org/wiki/Organisation_des_Nations_unies_pour_l%27alimentation_et_l%27agriculture, mais dans la base de données actuelle 175 

# Question 2 : Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique . 

Parmi les documents sur les Bilans alimentaires que vous avez téléchargés, il y a des informations redondantes. En effet, pour un pays donné, certaines de ces informations peuvent se calculer à partir d'autres :

Production 
Importations
Exportations 
Variation de stock 
Disponibilité intérieure 
Semences 
Pertes 
Nourriture, aussi appelée Disponibilité alimentaire 
Aliments pour animaux 
Traitement
Autres utilisations 

Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique (pas besoin de coder ici :soleil: ). C'est une équation à 3 termes de type \(a_1 + a2 + [...] = b_1 + b_2 + [...] = c_1 + c_2 + [...]\) ) faisant intervenir chacune des 11 quantités données ci dessus. Illustrez cette équation avec l'exemple du blé en France. Pour avoir un indice, 

In [101]:
 #Création d'un dataframe "FRANCE / BLE" pour vérifier les données
df_blé_France= df_vegetaux [(df_vegetaux['Zone']== 'France') & (df_vegetaux['Produit']== 'Blé')]
df_blé_France

Code Domaine                                            Domaine  \
37487         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37488         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37489         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37490         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37491         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37492         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37493         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37494         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37495         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37496         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37497         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37498         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37499         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37500         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37501         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

       Code zone    Zone  Code Élément  \
37487         68  France          5511   
37488         68  France          5611   
37489         68  France          5072   
37490         68  France          5911   
37491         68  France          5301   
37492         68  France          5521   
37493         68  France          5527   
37494         68  France          5123   
37495         68  France          5131   
37496         68  France          5154   
37497         68  France          5142   
37498         68  France           645   
37499         68  France           664   
37500         68  France           674   
37501         68  France           684   

                                                 Élément  Code Produit  \
37487                                         Production          2511   
37488                            Importations - Quantité          2511   
37489                                 Variation de stock          2511   
37490                            Exportations - Quantité          2511   
37491                           Disponibilité intérieure          2511   
37492                              Aliments pour animaux          2511   
37493                                           Semences          2511   
37494                                             Pertes          2511   
37495                                         Traitement          2511   
37496              Autres utilisations (non alimentaire)          2511   
37497                                         Nourriture          2511   
37498  Disponibilité alimentaire en quantité (kg/pers...          2511   
37499     Disponibilité alimentaire (Kcal/personne/jour)          2511   
37500  Disponibilité de protéines en quantité (g/pers...          2511   
37501  Disponibilité de matière grasse en quantité (g...          2511   

      Produit  Code année  Année               Unité    Valeur Symbole  \
37487     Blé        2013   2013  Milliers de tonnes  38614.00       S   
37488     Blé        2013   2013  Milliers de tonnes   2055.00       S   
37489     Blé        2013   2013  Milliers de tonnes   1131.00       S   
37490     Blé        2013   2013  Milliers de tonnes  21502.00       S   
37491     Blé        2013   2013  Milliers de tonnes  20298.00       S   
37492     Blé        2013   2013  Milliers de tonnes   7822.00       S   
37493     Blé        2013   2013  Milliers de tonnes    748.00       S   
37494     Blé        2013   2013  Milliers de tonnes    358.00       S   
37495     Blé        2013   2013  Milliers de tonnes   1575.00       S   
37496     Blé        2013   2013  Milliers de tonnes   2824.00       S   
37497     Blé        2013   2013  Milliers de tonnes   6971.00       S   
37498     Blé        2013   2013                  kg    108.43      Fc   
37499     Blé    

In [102]:
df_blé_France.shape

(15, 14)

# Une équation à 3 termes 

Disponibilité intérieur = Production + importations - exportations + variations des stocks = Nourriture + Aliments pour animaux + Semences + Traitement + Autres utilisations + Pertes

In [103]:
#Disponibilité intérieur
dispo_interieur1 = int(df_blé_France[df_blé_France['Élément'] == 'Disponibilité intérieure'].Valeur)
dispo_interieur1

20298

In [104]:
#Production + importations - exportations + variations des stocks 
dispo_interieur2= (int(df_blé_France[df_blé_France['Code Élément'] == 5511].Valeur) +
                   int(df_blé_France[df_blé_France['Code Élément'] == 5611].Valeur)-
                   int(df_blé_France[df_blé_France['Code Élément'] == 5911].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5072].Valeur))
dispo_interieur1

20298

In [105]:
#Nourriture + Aliments pour animaux + Semences + Traitement + Autres utilisations + Pertes
dispo_interieur3= (int(df_blé_France[df_blé_France['Code Élément'] == 5142].Valeur) +
                   int(df_blé_France[df_blé_France['Code Élément'] == 5521].Valeur)+
                   int(df_blé_France[df_blé_France['Code Élément'] == 5527].Valeur)+
                   int(df_blé_France[df_blé_France['Code Élément'] == 5131].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5154].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5123].Valeur))
dispo_interieur2

20298

### Réponse à la deuxième question :
Ainsi, les trois termes sont exactement équivalents dans les données France/Blé, et l'équation a été vérifiée.
    
        
            
                 
                      








# Question 3 : Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal puis en kg de protéines.

Vous ferez cela à partir de ces informations :

Population de chaque pays ;
Disponibilité alimentaire donnée pour chaque produit et pour chaque pays en kcal/personne/jour.
Disponibilité alimentaire en protéines donnée pour chaque produit et pour chaque pays en g/personne/jour.

In [106]:
#Ajouter une colonne 'Radical' aux deux tables df_vegetaux et df_animaux
df_vegetaux['Radical'] = 'plante'
df_animaux['Radical'] = 'animal'

In [107]:
df_animaux.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole', 'Radical'],
      dtype='object')

In [108]:
df_vegetaux.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole', 'Radical'],
      dtype='object')

Les colonnes sont identiques, on peut donc fusionner les deux tables

In [109]:
#fusionner les deux tables
df_vegetaux_animaux = df_animaux.append (df_vegetaux) 
df_vegetaux_animaux.sample(5)

Code Domaine                                            Domaine  \
45974         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
91222         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
56531         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
2577          FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
62257         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

       Code zone          Zone  Code Élément  \
45974         25  Îles Salomon           664   
91222        198      Slovénie          5072   
56531        119      Lettonie          5301   
2577          79     Allemagne          5611   
62257        132      Maldives          5611   

                                              Élément  Code Produit  \
45974  Disponibilité alimentaire (Kcal/personne/jour)          2571   
91222                              Variation de stock          2531   
56531                        Disponibilité intérieure          2542   
2577                          Importations - Quantité          2555   
62257                         Importations - Quantité          2641   

               Produit  Code année  Année               Unité  Valeur Symbole  \
45974    Huile de Soja        2013   2013  Kcal/personne/jour     0.0      Fc   
91222  Pommes de Terre        2013   2013  Milliers de tonnes    30.0       S   
56531    Sucre Eq Brut        2013   2013  Milliers de tonnes    76.0       S   
2577              Soja        2013   2013  Milliers de tonnes  3620.0       S   
62257          Piments        2013   2013  Milliers de tonnes     0.0       S   

      Description du Symbole Radical  
45974        Donnée calculée  plante  
91222  Données standardisées  plante  
56531  Données standardisées  plante  
2577   Données standardisées  plante  
62257  Données standardisées  plante

In [110]:
df_vegetaux_animaux.shape

(140919, 15)

In [111]:
df_vegetaux_animaux['Valeur'].describe()

count    140919.000000
mean        233.524210
std        4885.276773
min      -39863.000000
25%           0.000000
50%           1.000000
75%          15.000000
max      739267.000000
Name: Valeur, dtype: float64

In [112]:
# Jointure avec la table population sur le Code zone 
df_vegetaux_animaux= pd.merge(df_vegetaux_animaux,df_population, on = 'Code zone' )
df_vegetaux_animaux.sample(8)

Code Domaine_x                                          Domaine_x  \
119367           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
106047           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
113333           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
87052            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
15312            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
118096           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
69186            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
122215           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

        Code zone      Zone_x  Code Élément_x                 Élément_x  \
119367        195     Sénégal            5911   Exportations - Quantité   
106047        173     Pologne            5911   Exportations - Quantité   
113333        183    Roumanie            5123                    Pertes   
87052         136  Mauritanie            5301  Disponibilité intérieure   
15312          23      Belize            5511                Production   
118096        244       Samoa            5611   Importations - Quantité   
69186         110       Japon            5911   Exportations - Quantité   
122215        199   Slovaquie            5511                Production   

        Code Produit_x                Produit_x  Code année_x  Année_x  ...  \
119367            2532                   Manioc          2013     2013  ...   
106047            2572         Huile d'Arachide          2013     2013  ...   
113333            2740             Beurre, Ghee          2013     2013  ...   
87052             2605          Légumes, Autres          2013     2013  ...   
15312             2586  Huil Plantes Oleif Autr          2013     2013  ...   
118096            2767       Mollusques, Autres          2013     2013  ...   
69186             2766             Cephalopodes          2013     2013  ...   
122215            2657       Boissons Fermentés          2013     2013  ...   

       Code Élément_y          Élément_y Code Produit_y   Produit_y  \
119367            511  Population totale           2501  Population   
106047            511  Population totale           2501  Population   
113333            511  Population totale           2501  Population   
87052             511  Population totale           2501  Population   
15312             511  Population totale           2501  Population   
118096            511  Population totale           2501  Population   
69186             511  Population totale           2501  Population   
122215            511  Population totale           2501  Population   

       Code année_y Année_y         Unité_y Valeur_y  Symbole_y  \
119367         2013    2013  1000 personnes    14133        NaN   
106047         2013    2013  1000 personnes    38217        NaN   
113333         2013    2013  1000 personnes    21699        NaN   
87052          2013    2013  1000 personnes     3890        NaN   
15312          2013    2013  1000 personnes      332        NaN   
118096         2013    2013  1000 personnes      190        NaN   
69186          2013    2013  1000 personnes   127144        NaN   
122215         2013    2013  1000 personnes     5450        NaN   

       Description du Symbole_y  
119367        Donnée officielle  
106047        Donnée officielle  
113333        Donnée officielle  
87052         Donnée officielle  
15312         Donnée officielle  
118096        Donnée officielle  
69186         Donnée officielle  
122215        Donnée officielle  

[8 rows x 28 columns]

In [113]:
df_vegetaux_animaux.columns

Index(['Code Domaine_x', 'Domaine_x', 'Code zone', 'Zone_x', 'Code Élément_x',
       'Élément_x', 'Code Produit_x', 'Produit_x', 'Code année_x', 'Année_x',
       'Unité_x', 'Valeur_x', 'Symbole_x', 'Description du Symbole_x',
       'Radical', 'Code Domaine_y', 'Domaine_y', 'Zone_y', 'Code Élément_y',
       'Élément_y', 'Code Produit_y', 'Produit_y', 'Code année_y', 'Année_y',
       'Unité_y', 'Valeur_y', 'Symbole_y', 'Description du Symbole_y'],
      dtype='object')

In [114]:
#  recréer un DataFrame avec les colonnes nécessaires
df_vegetaux_animaux= df_vegetaux_animaux[['Code Domaine_x', 'Domaine_x', 'Code zone', 'Zone_x', 'Code Élément_x',
       'Élément_x', 'Code Produit_x', 'Produit_x', 'Année_x',
       'Unité_x', 'Valeur_x','Radical', 'Valeur_y']]

In [115]:
df_vegetaux_animaux.head()

Code Domaine_x                                          Domaine_x  \
0           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
1           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
2           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
3           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
4           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

   Code zone       Zone_x  Code Élément_x  \
0          2  Afghanistan            5511   
1          2  Afghanistan            5611   
2          2  Afghanistan            5301   
3          2  Afghanistan            5142   
4          2  Afghanistan             645   

                                           Élément_x  Code Produit_x  \
0                                         Production            2731   
1                            Importations - Quantité            2731   
2                           Disponibilité intérieure            2731   
3                                         Nourriture            2731   
4  Disponibilité alimentaire en quantité (kg/pers...            2731   

          Produit_x  Année_x             Unité_x  Valeur_x Radical  Valeur_y  
0  Viande de Bovins     2013  Milliers de tonnes    134.00  animal     30552  
1  Viande de Bovins     2013  Milliers de tonnes      6.00  animal     30552  
2  Viande de Bovins     2013  Milliers de tonnes    140.00  animal     30552  
3  Viande de Bovins     2013  Milliers de tonnes    140.00  animal     30552  
4  Viande de Bovins     2013                  kg      4.59  animal     30552

In [116]:
#Changer les noms de colonnes
df_vegetaux_animaux = df_vegetaux_animaux.rename(columns= {'Code Domaine_x':'Code Domaine', 'Domaine_x':'Domaine', 'Code zone': 'Code zone', 'Zone_x': 'Zone', 'Code Élément_x': 'Code Élément',
       'Élément_x':'Élément', 'Code Produit_x':'Code Produit', 'Produit_x': 'Produit', 'Année_x': 'Année',
       'Unité_x':'Unité', 'Valeur_x':'Valeur','Radical':'Radical', 'Valeur_y':'Population'})   

In [117]:
df_vegetaux_animaux.head()


Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément  \
0  Afghanistan          5511   
1  Afghanistan          5611   
2  Afghanistan          5301   
3  Afghanistan          5142   
4  Afghanistan           645   

                                             Élément  Code Produit  \
0                                         Production          2731   
1                            Importations - Quantité          2731   
2                           Disponibilité intérieure          2731   
3                                         Nourriture          2731   
4  Disponibilité alimentaire en quantité (kg/pers...          2731   

            Produit  Année               Unité  Valeur Radical  Population  
0  Viande de Bovins   2013  Milliers de tonnes  134.00  animal       30552  
1  Viande de Bovins   2013  Milliers de tonnes    6.00  animal       30552  
2  Viande de Bovins   2013  Milliers de tonnes  140.00  animal       30552  
3  Viande de Bovins   2013  Milliers de tonnes  140.00  animal       30552  
4  Viande de Bovins   2013                  kg    4.59  animal       30552

In [118]:
#Population par pays
df_vegetaux_animaux['Population'] = df_vegetaux_animaux['Population'] * 1000

In [119]:
df_vegetaux_animaux.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Année', 'Unité', 'Valeur',
       'Radical', 'Population'],
      dtype='object')

In [120]:
df_vegetaux_animaux['Valeur'].describe()

count    140919.000000
mean        233.524210
std        4885.276773
min      -39863.000000
25%           0.000000
50%           1.000000
75%          15.000000
max      739267.000000
Name: Valeur, dtype: float64

# Observation:
Il y a des valeurs inférieures à zéro, ça n'a pas de sens, donc ça devrait être supprimé mais je ne savais pas

In [121]:
# On créé un pivot avec les éléments en colonne
df_vegetaux_animaux = df_vegetaux_animaux.pivot_table( index = [ 'Code zone', 'Zone', 'Code Produit', 'Produit', 'Année', 'Radical', 'Population'] , columns = 'Élément', values = 'Valeur', aggfunc = sum).reset_index()

In [122]:
df_vegetaux_animaux.head()

Élément  Code zone     Zone  Code Produit Produit  Année Radical  Population  \
0                1  Arménie          2511     Blé   2013  plante     2977000   
1                1  Arménie          2513    Orge   2013  plante     2977000   
2                1  Arménie          2514    Maïs   2013  plante     2977000   
3                1  Arménie          2515  Seigle   2013  plante     2977000   
4                1  Arménie          2516  Avoine   2013  plante     2977000   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
0                         93.0                                    0.0   
1                        137.0                                   26.0   
2                         96.0                                    NaN   
3                          1.0                                    NaN   
4                          4.0                                    NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  \
0                                                1024.0  ...   
1                                                   0.0  ...   
2                                                   0.0  ...   
3                                                   1.0  ...   
4                                                   2.0  ...   

Élément  Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                    30.52          
1                                                     0.00          
2                                                     0.01          
3                                                     0.02          
4                                                     0.09          

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                           554.0                      1.0   
1                           198.0                      0.0   
2                           102.0                      NaN   
3                             1.0                      NaN   
4                             6.0                      NaN   

Élément  Importations - Quantité  Nourriture  Pertes  Production  Semences  \
0                          361.0       389.0    32.0       312.0      30.0   
1                            9.0         0.0    15.0       189.0      14.0   
2                           82.0         0.0     7.0        21.0       0.0   
3                            0.0         0.0     0.0         1.0       0.0   
4                            1.0         1.0     0.0         5.0       0.0   

Élément  Traitement  Variation de stock  
0              10.0              -118.0  
1               7.0                 0.0  
2               NaN                 NaN  
3               NaN                 0.0  
4               NaN                 NaN  

[5 rows x 22 columns]

In [123]:
    # Calculs des disponibilités alimentaires totales par produit pour chaque pays (Kg) = Population * (Disponibilité de protéines en quantité (g/personne/jour))/1000 * 365
    df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)'] = (df_vegetaux_animaux['Population'] *df_vegetaux_animaux['Disponibilité de protéines en quantité (g/personne/jour)']/1000 * 365 )
    

In [124]:
# Calculs des disponibilités alimentaires totales par produit pour chaque pays(kcal) = Population * (Disponibilité de protéines en quantité Disponibilité alimentaire (Kcal/personne/jour) * 365
df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)'] = (df_vegetaux_animaux['Population'] *df_vegetaux_animaux['Disponibilité alimentaire (Kcal/personne/jour)'] * 365 )

In [125]:
df_vegetaux_animaux.sample(10)

Élément  Code zone          Zone  Code Produit                  Produit  \
631             10     Australie          2572         Huile d'Arachide   
4985            83      Kiribati          2602                  Oignons   
3811            60   El Salvador          2586  Huil Plantes Oleif Autr   
2235            38     Sri Lanka          2532                   Manioc   
15245          255      Belgique          2848       Lait - Excl Beurre   
7232           113  Kirghizistan          2765                Crustacés   
3215            53         Bénin          2743                    Crème   
3014            50        Chypre          2635                      Thé   
1248            20      Botswana          2532                   Manioc   
8851           136    Mauritanie          2732   Viande d'Ovins/Caprins   

Élément  Année Radical  Population  Aliments pour animaux  \
631       2013  plante    23343000                    NaN   
4985      2013  plante      102000                    NaN   
3811      2013  plante     6340000                    NaN   
2235      2013  plante    21273000                   76.0   
15245     2013  animal    11104000                  885.0   
7232      2013  animal     5548000                    NaN   
3215      2013  animal    10323000                    NaN   
3014      2013  plante     1141000                    NaN   
1248      2013  plante     2021000                    NaN   
8851      2013  animal     3890000                    NaN   

Élément  Autres utilisations (non alimentaire)  \
631                                        NaN   
4985                                       NaN   
3811                                      10.0   
2235                                      16.0   
15245                                    165.0   
7232                                       0.0   
3215                                       NaN   
3014                                       NaN   
1248                                       0.0   
8851                                       NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  \
631                                                 2.0  ...   
4985                                                0.0  ...   
3811                                               15.0  ...   
2235                                               41.0  ...   
15245                                             379.0  ...   
7232                                                0.0  ...   
3215                                                0.0  ...   
3014                                                0.0  ...   
1248                                                0.0  ...   
8851                                               46.0  ...   

Élément  Exportations - Quantité  Importations - Quantité  Nourriture  Pertes  \
631                          1.0                      1.0         2.0     NaN   
4985                         NaN                      0.0         0.0     NaN   
3811                         5.0                     19.0         4.0     NaN   
2235                        12.0                     24.0       207.0    15.0   
15245                     4337.0                   4332.0      2623.0     NaN   
7232                         0.0                      0.0         0.0     NaN   
3215                         NaN                      0.0         0.0     NaN   
3014                         0.0                      1.0         1.0     NaN   
1248                         NaN                      0.0         0.0     NaN   
8851                         NaN                      0.0        46.0     NaN   

Élément  Production  Semences  Traitement  Variation de stock  \
631             1.0       NaN         NaN                 NaN   
4985            NaN       NaN         NaN                 NaN   
3811            0.0       NaN         0.0                 0.0   
2235          303.0       NaN         NaN                 NaN   
15245        3483.0       NaN

In [126]:
df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)'].describe()

count    1.156100e+04
mean     1.792744e+07
std      1.857800e+08
min     -1.717080e+07
25%      3.942000e+03
50%      2.169341e+05
75%      2.510412e+06
max      8.865481e+09
Name: Disponibilité alimentaire protéines totale (Kg), dtype: float64

In [127]:
df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)'].describe()

count    1.424100e+04
mean     5.170903e+11
std      6.225858e+12
min     -9.745588e+11
25%      0.000000e+00
50%      7.594920e+09
75%      8.094605e+10
max      4.071142e+14
Name: Disponibilité alimentaire totale (Kcal), dtype: float64

#Résultat : Dans le monde, l'accès à la nourriture en quantité n'est pas égalitaire

# Question 4 : Ratios énergétiques
A partir de ces dernières informations, et à partir du poids de la disponibilité alimentaire (pour chaque pays et chaque produit), calculez pour chaque produit le ratio "énergie/poids", que vous donnerez en kcal/kg. Vous pouvez vérifier la cohérence de votre calcul en comparant ce ratio aux données disponibles sur internet, par exemple en cherchant la valeur calorique d'un oeuf.

Indication : La disponibilité alimentaire en kcal/personne/jour est calculée par la FAO en multipliant la quantité Nourriture par le ratio énergie/poids (en kcal/kg), puis en le divisant par la population du pays puis par 365. Ici, on vous demande juste de retrouver le ratio énergie/poids que la FAO a utilisé dans son calcul.

In [128]:
df_vegetaux_animaux['Nourriture'].replace(0, np.nan, inplace=True)

In [129]:
#Calcul du ratio énergie/poids (en kcal/kg)= La disponibilité alimentaire en kcal/kg / la quantité Nourriture
df_vegetaux_animaux['Ratio énergie/poids (kcal/kg)'] = (df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)']
                                                        / (df_vegetaux_animaux['Nourriture'] * 1000000)) 


In [130]:
df_vegetaux_animaux_Oeufs = df_vegetaux_animaux[df_vegetaux_animaux['Produit'] == 'Oeufs']
df_vegetaux_animaux_Oeufs .head()

Élément  Code zone         Zone  Code Produit Produit  Année Radical  \
74               1      Arménie          2744   Oeufs   2013  animal   
144              2  Afghanistan          2744   Oeufs   2013  animal   
226              3      Albanie          2744   Oeufs   2013  animal   
319              4      Algérie          2744   Oeufs   2013  animal   
400              7       Angola          2744   Oeufs   2013  animal   

Élément  Population  Aliments pour animaux  \
74          2977000                    NaN   
144        30552000                    NaN   
226         3173000                    NaN   
319        39208000                    NaN   
400        21472000                    NaN   

Élément  Autres utilisations (non alimentaire)  \
74                                         0.0   
144                                        NaN   
226                                        NaN   
319                                        NaN   
400                                        NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  \
74                                                 45.0  ...   
144                                                 5.0  ...   
226                                                46.0  ...   
319                                                27.0  ...   
400                                                 4.0  ...   

Élément  Importations - Quantité  Nourriture  Pertes  Production  Semences  \
74                           1.0        35.0     NaN        34.0       0.0   
144                         29.0        40.0     5.0        18.0       2.0   
226                          0.0        40.0     7.0        48.0       0.0   
319                          0.0       316.0    10.0       347.0      21.0   
400                         24.0        24.0     2.0         5.0       3.0   

Élément  Traitement  Variation de stock  \
74              NaN                 NaN   
144             NaN                 NaN   
226             NaN                 NaN   
319             NaN                 NaN   
400             NaN                 NaN   

Élément  Disponibilité alimentaire protéines totale (Kg)  \
74                                            3727055.15   
144                                           4237562.40   
226                                           4227229.25   
319                                          32915116.00   
400                                           2586302.40   

Élément  Disponibilité alimentaire totale (Kcal)  \
74                                  4.889722e+10   
144                                 5.575740e+10   
226                                 5.327467e+10   
319                                 3.863948e+11   
400                                 3.134912e+10   

Élément  Ratio énergie/poids (kcal/kg)  
74                         1397.063571  
144                        1393.935000  
226                        1331.866750  
319                        1222.768481  
400                        1306.213333  

[5 rows x 25 columns]

Les calories dans les œufs (155 kcal) en 100 g, soit 1550 par kg, selon Wikipedia https://fr.wikipedia.org/wiki/%C5%92uf_(aliment) 
Il est curieux de voir à quel point les ratios énergétiques diffèrent de la valeur attendue, mais on constate que la valeur moyenne est proche (à 1345 kcal/kg).

In [131]:
#le pourcentage de protéines de chaque produit (pour chaque pays).
df_vegetaux_animaux['Pourcentage de protéines produit'] = (df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)']  / (df_vegetaux_animaux['Nourriture'] * 1000000))*100



In [132]:
df_vegetaux_animaux[df_vegetaux_animaux['Produit'] == 'Avoine'].describe()

Élément   Code zone  Code Produit   Année    Population  \
count    167.000000         167.0   167.0  1.670000e+02   
mean     125.269461        2516.0  2013.0  4.097872e+07   
std       72.337265           0.0     0.0  1.484793e+08   
min        1.000000        2516.0  2013.0  5.400000e+04   
25%       66.500000        2516.0  2013.0  2.811500e+06   
50%      120.000000        2516.0  2013.0  9.357000e+06   
75%      184.500000        2516.0  2013.0  2.831300e+07   
max      276.000000        2516.0  2013.0  1.385567e+09   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
count               122.000000                              13.000000   
mean                133.204918                              15.846154   
std                 400.507672                              50.875413   
min                   0.000000                               0.000000   
25%                   0.000000                               0.000000   
50%                   2.500000                               1.000000   
75%                  83.000000                               4.000000   
max                3298.000000                             185.000000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
count                                        158.000000   
mean                                           5.569620   
std                                            9.643563   
min                                          -21.000000   
25%                                            0.000000   
50%                                            1.000000   
75%                                            8.000000   
max                                           60.000000   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                           157.000000        
mean                                              0.964522        
std                                               1.598133        
min                                              -1.930000        
25%                                               0.020000        
50%                                               0.220000        
75%                                               1.370000        
max                                               9.490000        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                           141.000000               
mean                                              0.108440               
std                                               0.173873               
min                                              -0.030000               
25%                                               0.000000               
50%                                               0.030000               
75%                                               0.150000               
max                                               1.120000               

Élément  Disponibilité de protéines en quantité (g/personne/jour)  ...  \
count                                           150.000000         ...   
mean                                              0.238000         ...   
std                                               0.384988         ...   
min                                              -0.370000         ...   
25%                                               0.010000         ...   
50%                                               0.065000         ...   
75%                                               0.320000         ...   
max                                               2.320000         ...   

Élément   Nourriture     Pertes   Production     Semences  Traitement  \
count      94.000000  65.000000    76.000000    71.000000    7.000000   
mean       41.521277   7.907692   313.368421    35.000000    5.142857   
std       148.970787  14.977820   754.603134   155.261991   13.171036   
min      -246.000000   0.000000     0.000000    

Le pourcentage de protéines dans l'avoine est Protéine.
16,9 g selon https://en.wikipedia.org/wiki/Oat#Nutrient_profile

In [133]:
df_vegetaux_animaux.describe()

Élément     Code zone  Code Produit    Année    Population  \
count    15605.000000  15605.000000  15605.0  1.560500e+04   
mean       125.334508   2631.376610   2013.0  4.219567e+07   
std         73.151148     91.753174      0.0  1.502135e+08   
min          1.000000   2511.000000   2013.0  5.400000e+04   
25%         63.000000   2558.000000   2013.0  2.839000e+06   
50%        120.000000   2613.000000   2013.0  9.511000e+06   
75%        188.000000   2732.000000   2013.0  2.893400e+07   
max        276.000000   2899.000000   2013.0  1.385567e+09   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
count              2720.000000                            5496.000000   
mean                479.501838                             157.391376   
std                4240.119637                            5076.785816   
min                   0.000000                               0.000000   
25%                   0.000000                               0.000000   
50%                   4.000000                               0.000000   
75%                  74.000000                               4.000000   
max              150000.000000                          347309.000000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
count                                      14241.000000   
mean                                          34.789832   
std                                          107.287655   
min                                          -21.000000   
25%                                            0.000000   
50%                                            4.000000   
75%                                           21.000000   
max                                         1711.000000   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                         14015.000000        
mean                                              8.719368        
std                                              24.618223        
min                                              -1.930000        
25%                                               0.060000        
50%                                               0.830000        
75%                                               5.190000        
max                                             430.760000        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                         11794.000000               
mean                                              1.283111               
std                                               3.680399               
min                                              -0.030000               
25%                                               0.010000               
50%                                               0.080000               
75%                                               0.630000               
max                                              60.760000               

Élément  Disponibilité de protéines en quantité (g/personne/jour)  ...  \
count                                         11561.000000         ...   
mean                                              1.223608         ...   
std                                               3.598686         ...   
min                                              -0.370000         ...   
25%                                               0.010000         ...   
50%                                               0.100000         ...   
75%                                               0.660000         ...   
max                                              54.970000         ...   

Élément     Nourriture        Pertes     Production      Semences  \
count      9997.000000   4278.000000    9180.000000   2091.000000   
mean        487.772132    106.053763    1090.379085     73.974653   
std        5292.991932   1113.100416   12067.344094    528.069224   
min        -246.000000      0.000000 

#### L’Agence considère que la référence nutritionnelle en protéines des adultes en bonne santé (RNP) est de 0,83 g/kg/j.. La moyenne arithmétique indique que les humains obtiennent leurs besoins caloriques. Mais en examinant l'écart type et la valeur la plus élevée et la plus faible, nous concluons que la répartition des produits riches en calories et en protéines est inégale d'un pays à l'autre.

# Question 5 :

Citez 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids.
Étonnamment, il arrive que ce ratio soit différent en fonction du pays. Il faudra donc réaliser pour chaque aliment une moyenne sur les différents pays. Vous créerez donc une nouvelle table grâce à une agrégation. Attention à bien retirer les valeurs égales à 0 afin de ne pas fausser le calcul de la moyenne.
Citez 5 aliments parmi les 20 aliments  les plus caloriques et les plus riches en protéines.




In [134]:
# Créer un nouveau DataFrame avec  les valaurs supérieur à 0

df_grandes_calorique = df_vegetaux_animaux[df_vegetaux_animaux['Ratio énergie/poids (kcal/kg)'] > 0]



In [135]:
# Puis on applique un group by sur le Produit 
df_grandes_calorique = df_grandes_calorique.groupby('Produit').mean().reset_index()
df_grandes_calorique.head()

Élément                   Produit   Code zone  Code Produit   Année  \
0                Abats Comestible  127.664596        2736.0  2013.0   
1                 Agrumes, Autres  132.459459        2614.0  2013.0   
2           Aliments pour enfants  125.843373        2680.0  2013.0   
3                          Ananas  121.652893        2618.0  2013.0   
4        Animaux Aquatiques Autre  117.500000        2769.0  2013.0   

Élément    Population  Aliments pour animaux  \
0        4.344527e+07              45.863636   
1        6.133273e+07                    NaN   
2        3.488387e+07                    NaN   
3        4.919470e+07                    NaN   
4        3.772612e+08                    NaN   

Élément  Autres utilisations (non alimentaire)  \
0                                    16.307692   
1                                     3.000000   
2                                          NaN   
3                                     0.428571   
4                                     0.000000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
0                                              8.894410   
1                                              3.351351   
2                                              6.024096   
3                                              5.371901   
4                                              1.000000   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                 2.853106        
1                                                 4.808108        
2                                                 0.593614        
3                                                 5.281405        
4                                                 0.747500        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  ...  \
0                                                 0.264099              ...   
1                                                 0.022941              ...   
2                                                 0.047470              ...   
3                                                 0.017273              ...   
4                                                 0.007500              ...   

Élément  Nourriture     Pertes  Production  Semences  Traitement  \
0         97.322981   3.476190  115.472050       NaN         NaN   
1        275.513514  26.653846  399.307692       NaN    0.400000   
2          8.819277   1.000000         NaN       NaN         NaN   
3        170.322314  32.294118  367.530303       NaN    0.888889   
4        265.750000        NaN  393.333333       0.0         NaN   

Élément  Variation de stock  Disponibilité alimentaire protéines totale (Kg)  \
0                  0.500000                                     1.738781e+07   
1                  1.933333                                     1.264902e+06   
2                  0.000000                                     1.362014e+06   
3                  6.345455                                     4.613769e+05   
4                  0.000000                                     1.005391e+07   

Élément  Disponibilité alimentaire totale (Kcal)  \
0                                   1.070618e+11   
1                                   6.685396e+10   
2                                   3.340675e+10   
3                                   5.848916e+10   
4                                   1.377004e+11   

Élément  Ratio énergie/poids (kcal/kg)  Pourcentage de protéines produit  
0                          1129.258163                         17.561047  
1                           248.981045                          0.388122  
2                          3690.832338                         14.947422  
3                           439.413583                          0.282031  
4                           666.357729                          4.504852  

[5 rows x 24 columns]

In [136]:
#Trier pour obtenir les produits les plus caloriques
df_grandes_calorique.sort_values(by=['Ratio énergie/poids (kcal/kg)'], inplace=True, ascending=False)
df_grandes_calorique.head(20)

Élément                   Produit   Code zone  Code Produit   Année  \
43       Huiles de Foie de Poisso  193.000000        2782.0  2013.0   
41                Huile de Sésame  119.200000        2579.0  2013.0   
40            Huile de Son de Riz  119.166667        2581.0  2013.0   
44             Huiles de Poissons   77.333333        2781.0  2013.0   
32               Huile d'Arachide  129.328947        2572.0  2013.0   
33                  Huile d'Olive  126.235294        2580.0  2013.0   
36         Huile de Germe de Maïs  131.269231        2582.0  2013.0   
30        Huil Plantes Oleif Autr  131.041667        2586.0  2013.0   
38             Huile de Palmistes  138.346939        2576.0  2013.0   
31         Huile Graines de Coton  130.555556        2575.0  2013.0   
37                 Huile de Palme  128.514286        2577.0  2013.0   
42             Huile de Tournesol  128.685484        2573.0  2013.0   
39                  Huile de Soja  124.940299        2571.0  2013.0   
35        Huile de Colza&Moutarde  128.348485        2574.0  2013.0   
34                  Huile de Coco  135.397260        2578.0  2013.0   
28         Graisses Animales Crue  127.882759        2737.0  2013.0   
9                    Beurre, Ghee  126.968992        2740.0  2013.0   
60                      Palmistes   39.000000        2562.0  2013.0   
84                         Sésame  129.898551        2561.0  2013.0   
5          Arachides Decortiquees  129.220588        2556.0  2013.0   

Élément    Population  Aliments pour animaux  \
43       2.041050e+07                    NaN   
41       1.173457e+08                    NaN   
40       2.863371e+08                    NaN   
44       6.568467e+07                  21.50   
32       7.243880e+07                    NaN   
33       4.734812e+07                   0.00   
36       5.508121e+07                    NaN   
30       4.624730e+07                   0.00   
38       7.860473e+07                    NaN   
31       8.473535e+07                    NaN   
37       5.593403e+07                   0.00   
42       4.956136e+07                    NaN   
39       5.022874e+07                   2.75   
35       7.668285e+07                 700.00   
34       7.583960e+07                    NaN   
28       4.798956e+07                 155.60   
9        5.197145e+07                   1.00   
60       1.335100e+07                    NaN   
84       7.676384e+07                    NaN   
5        4.733350e+07                   1.00   

Élément  Autres utilisations (non alimentaire)  \
43                                    0.000000   
41                                   57.200000   
40                                   66.000000   
44                                    4.000000   
32                                   47.916667   
33                                    6.098039   
36                                    6.727273   
30                                   29.554622   
38                                  135.333333   
31                                   33.419355   
37                                  277.171429   
42                                   46.070423   
39                                  245.915493   
35                                  278.272727   
34                                   20.230769   
28                                   76.892857   
9                                     1.400000   
60                                         NaN   
84                                    6.333333   
5                                    70.277778   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
43                                            10.000000   
41                                             9.314286   
40                                             7.916667   
44                                            34.000000   
32                                            26.000000   
33                                            28.188235   
36                                 

#Réponse à la question \
Les 5 aliments parmi les 20 aliments les plus caloriques\
Huiles de Foie de Poisso\
Huile de Sésame\
Huile de Son de Riz\
Huiles de Poissons \
Huile d'Arachide

In [137]:
# Créer un nouveau DataFrame avec  les valaurs supérieur à 0

df_grandes_protéines= df_vegetaux_animaux[df_vegetaux_animaux['Pourcentage de protéines produit'] > 0]

# Puis on applique un group by sur le Produit 
df_grandes_protéines = df_grandes_protéines.groupby('Produit').mean().reset_index()


#Trier pour obtenir les produits les plus caloriques
df_grandes_protéines.sort_values(by=['Pourcentage de protéines produit'], inplace=True, ascending=False)
df_grandes_protéines.head(20)

Élément                   Produit   Code zone  Code Produit   Année  \
26         Graines Colza/Moutarde   33.000000        2558.0  2013.0   
73                           Soja  134.626506        2555.0  2013.0   
5          Arachides Decortiquees  127.615942        2556.0  2013.0   
62                           Pois  134.162162        2547.0  2013.0   
44            Légumineuses Autres  128.507576        2549.0  2013.0   
29                       Haricots  129.777778        2546.0  2013.0   
0                Abats Comestible  127.664596        2736.0  2013.0   
78                         Sésame  129.616438        2561.0  2013.0   
85                  Viande, Autre  131.976923        2735.0  2013.0   
61       Plantes Oleiferes, Autre  146.428571        2570.0  2013.0   
2           Aliments pour enfants  130.500000        2680.0  2013.0   
82               Viande de Bovins  125.514620        2731.0  2013.0   
15                   Cephalopodes  128.902778        2766.0  2013.0   
81         Viande d'Ovins/Caprins  127.477124        2732.0  2013.0   
84            Viande de Volailles  125.431034        2734.0  2013.0   
59                        Piments  140.081395        2641.0  2013.0   
65            Poissons Pelagiques  123.958084        2763.0  2013.0   
27           Graines de tournesol  136.000000        2557.0  2013.0   
64        Poissons Marins, Autres  127.160839        2764.0  2013.0   
87                 Épices, Autres  129.669421        2645.0  2013.0   

Élément    Population  Aliments pour animaux  \
26       3.518200e+07             739.000000   
73       7.250643e+07             517.000000   
5        4.923415e+07               1.000000   
62       5.691562e+07              94.200000   
44       5.059539e+07              95.746269   
29       5.099910e+07              90.615385   
0        4.344527e+07              45.863636   
78       7.527011e+07                    NaN   
85       5.250905e+07               6.222222   
61       4.552057e+07              13.642857   
2        3.944504e+07                    NaN   
82       4.091769e+07               3.000000   
15       6.024976e+07               7.800000   
81       4.542072e+07               9.000000   
84       4.021452e+07               0.000000   
59       7.150685e+07                    NaN   
65       4.064465e+07             123.677632   
27       4.299121e+07              94.166667   
64       4.138161e+07              32.642857   
87       5.596300e+07                    NaN   

Élément  Autres utilisations (non alimentaire)  \
26                                         NaN   
73                                   90.428571   
5                                    70.277778   
62                                    5.600000   
44                                   40.800000   
29                                    9.833333   
0                                    16.307692   
78                                    6.333333   
85                                    1.363636   
61                                    5.515152   
2                                          NaN   
82                                    2.514286   
15                                    0.319444   
81                                    4.200000   
84                                   31.450000   
59                                   85.000000   
65                                    2.083832   
27                                    0.000000   
64                                    1.174825   
87                                   18.200000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
26                                            16.000000   
73                                            12.771084   
5                                             25.253623   
62                                            12.000000   
44                                            36.393939   
29                                            34.619048   
0                                  

 LES 5 aliments parmi les 20 aliments les plus riches en protéines.\
Graines Colza/Moutarde
Soja\
Arachides Decortiquees\
Pois\
Légumineuses Autres

# Question 6 : Calculez, pour les produits végétaux uniquement, la disponibilité intérieure mondiale exprimée en kcal.

In [138]:
df_vegetaux_animaux['Radical'].unique()

array(['plante', 'animal'], dtype=object)

In [139]:
# Création d'un nouveau DataFrame contenant uniquement les végétaux
df_végétaux_dispo_intérieure = df_vegetaux_animaux[df_vegetaux_animaux['Radical'] == 'plante']



In [140]:
df_végétaux_dispo_intérieure['Dispo intérieure (Kcal)'] = (df_végétaux_dispo_intérieure['Disponibilité intérieure'] * 1000000  *  df_végétaux_dispo_intérieure['Ratio énergie/poids (kcal/kg)'])



<ipython-input-140-2c5886c28da6>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_végétaux_dispo_intérieure['Dispo intérieure (Kcal)'] = (df_végétaux_dispo_intérieure['Disponibilité intérieure'] * 1000000  *  df_végétaux_dispo_intérieure['Ratio énergie/poids (kcal/kg)'])


In [141]:
df_végétaux_dispo_intérieure['Dispo intérieure protéines'] = ((df_végétaux_dispo_intérieure['Disponibilité intérieure'] * 1000000)  *  (df_végétaux_dispo_intérieure['Pourcentage de protéines produit' ]/100))


<ipython-input-141-c12905129eae>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_végétaux_dispo_intérieure['Dispo intérieure protéines'] = ((df_végétaux_dispo_intérieure['Disponibilité intérieure'] * 1000000)  *  (df_végétaux_dispo_intérieure['Pourcentage de protéines produit' ]/100))


In [142]:
df_végétaux_dispo_intérieure_produits

NameError: name 'df_végétaux_dispo_intérieure_produits' is not defined

In [143]:
df_végétaux_dispo_intérieure_produits = df_végétaux_dispo_intérieure.groupby(by = 'Produit').sum().reset_index()

In [144]:
df_végétaux_dispo_intérieure_produits

Élément                 Produit  Code zone  Code Produit   Année  Population  \
0               Agrumes, Autres      20116        418240  322080  6827946000   
1        Alcool, non Comestible      20111        425440  322080  6740769000   
2         Aliments pour enfants      21379        458280  344223  6962149000   
3                        Ananas      21118        439824  338184  6859134000   
4        Arachides Decortiquees      21372        434520  342210  6934565000   
..                          ...        ...           ...     ...         ...   
70                       Sésame      20191        414882  326106  6960363000   
71                          Thé      21709        455855  348249  6972431000   
72                      Tomates      21588        449973  348249  6905646000   
73                          Vin      21591        456660  346236  6969062000   
74               Épices, Autres      21496        454940  346236  6967191000   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
0                          0.0                                   25.0   
1                          0.0                                21769.0   
2                          0.0                                    0.0   
3                          0.0                                    3.0   
4                          1.0                                 1265.0   
..                         ...                                    ...   
70                         3.0                                   63.0   
71                         0.0                                   26.0   
72                      1870.0                                   77.0   
73                         0.0                                 1588.0   
74                         0.0                                  185.0   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
0                                                 128.0   
1                                                   0.0   
2                                                 652.0   
3                                                 659.0   
4                                                3662.0   
..                                                  ...   
70                                                736.0   
71                                                123.0   
72                                               1634.0   
73                                               2023.0   
74                                               1015.0   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                   209.07        
1                                                     0.00        
2                                                    65.14        
3                                                   653.61        
4                                                   241.59        
..                                                     ...        
70                                                   46.78        
71                                                  127.94        
72                                                 3115.94        
73                                                 1059.75        
74                                                  107.09        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  ...  \
0                                                     0.78              ...   
1                                                     0.00              ...   
2                                                     5.06              ...   
3                                                     1.90              ...   
4                                                   304.42              ...   
..                                                     ...              ...   
70                                                   64.47              ...   
71              

In [145]:
disponibilité_intérieure_mondiale_kcal = df_végétaux_dispo_intérieure_produits['Dispo intérieure (Kcal)'].sum()

In [146]:
disponibilité_intérieure_mondiale_protéines = df_végétaux_dispo_intérieure_produits['Dispo intérieure protéines'].sum()

In [147]:
print( 'La disponibilité intérieure mondiale des  produits végétaux est ','{:,}'.format(int(disponibilité_intérieure_mondiale_kcal)).replace(',', ' ') , ' kcal' )

La disponibilité intérieure mondiale des  produits végétaux est  12 312 091 265 143 276  kcal


In [148]:
print( 'La disponibilité intérieure mondiale des  produits végétaux est ','{:,}'.format(int(disponibilité_intérieure_mondiale_protéines)).replace(',', ' ') , ' protéines' )

La disponibilité intérieure mondiale des  produits végétaux est  290 177 334 142  protéines


Réponse 6: La disponibilité interieure mondiale des produits végétaux représente au moins 12 quatrillion ou 12 mille-billions de kcal.

# Question 7 :
Combien d'humains pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

Calculez combien d'humains pourraient être nourris si tous les produits végétaux disponibles localement dans le monde étaient utilisés pour l'alimentation en termes de calories.\
Selon https://fr.wikipedia.org/wiki/Ration_alimentaire, une personne normale a besoin de 2500 calories par jour

In [149]:
#Calculez combien d'humains pourraient être nourris si tous les produits végétaux disponibles localement dans le monde étaient utilisés pour l'alimentation en termes de calories.

Énergieune_personne_besoin_jour = 2500  
Énergieune_personne_besoin_an = Énergieune_personne_besoin_jour * 365
énergie_suffisante = disponibilité_intérieure_mondiale_kcal/ Énergieune_personne_besoin_an
print('Le nombre de personnes qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en caloriesest ',"{0:,}".format(énergie_suffisante));

print('Le nombre de personnes qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en  protéines est  ',énergie_suffisante/nb_population * 100, '%')


Le nombre de personnes qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en caloriesest  13,492,702,756.321398
Le nombre de personnes qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en  protéines est   192.8265562633697 %


Calculez combien d'humains pourraient être nourris si tous les produits végétaux disponibles localement dans le monde étaient utilisés pour l'alimentation en termes de calories protéiques.\
Selon https://www.cerin.org/breves-scientifiques/actualites-en-bref-et-articles-selectionnes-pour-vous-nutri-doc-n118/, une personne normale a besoin de 55 g de protéines par jour.

In [150]:
nb_besoin_protéique_journalier = 55.    
nb_besoin_protéique_kg_an = (nb_besoin_protéique_journalier/1000) * 365
nb_protéines_suffisante =  disponibilité_intérieure_mondiale_protéines / nb_besoin_protéique_kg_an
print('Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en calories est',
      "{0:,}".format(nb_protéines_suffisante));
print( 'pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en  protéines est'
      ,nb_protéines_suffisante/nb_population * 100, '%')

Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en calories est 14,454,661,725.63563
pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture en  protéines est 206.57407880718478 %


# Question 8 : 
Combien d'humains pourraient être nourris si toute la disponibilité alimentaire en produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [151]:
df_végétaux_dispo_intérieure.columns

Index(['Code zone', 'Zone', 'Code Produit', 'Produit', 'Année', 'Radical',
       'Population', 'Aliments pour animaux',
       'Autres utilisations (non alimentaire)',
       'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité intérieure', 'Exportations - Quantité',
       'Importations - Quantité', 'Nourriture', 'Pertes', 'Production',
       'Semences', 'Traitement', 'Variation de stock',
       'Disponibilité alimentaire protéines totale (Kg)',
       'Disponibilité alimentaire totale (Kcal)',
       'Ratio énergie/poids (kcal/kg)', 'Pourcentage de protéines produit',
       'Dispo intérieure (Kcal)', 'Dispo intérieure protéines'],
      dtype='object', name='Élément')

In [152]:
#Ajout de deux colonnes au dataframe df_dispo_interieure_vegetaux pour les calculs Q8

df_végétaux_dispo_intérieure['Dispo végétaux_animaux_pertes(Kcal)'] = ((df_végétaux_dispo_intérieure['Disponibilité alimentaire totale (Kcal)'])
                                                               + ((df_végétaux_dispo_intérieure['Aliments pour animaux'] * 1000000) 
                                                                  * df_végétaux_dispo_intérieure['Ratio énergie/poids (kcal/kg)'])
                                                               + ((df_végétaux_dispo_intérieure['Pertes'] * 1000000) 
                                                                  * df_végétaux_dispo_intérieure['Ratio énergie/poids (kcal/kg)'])
                                                              )
df_végétaux_dispo_intérieure['Dispo végétaux_animaux_pertes protéines(Kg)'] = (df_végétaux_dispo_intérieure['Disponibilité alimentaire protéines totale (Kg)']
                                                                      + (((df_végétaux_dispo_intérieure['Aliments pour animaux'] 
                                                                         + df_végétaux_dispo_intérieure['Pertes']) * 1000000)
                                                                       * (df_végétaux_dispo_intérieure['Pourcentage de protéines produit']/100))
                                                                      )




df_végétaux_dispo_intérieure.head()


<ipython-input-152-decfd514ce2e>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_végétaux_dispo_intérieure['Dispo végétaux_animaux_pertes(Kcal)'] = ((df_végétaux_dispo_intérieure['Disponibilité alimentaire totale (Kcal)'])
<ipython-input-152-decfd514ce2e>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_végétaux_dispo_intérieure['Dispo végétaux_animaux_pertes protéines(Kg)'] = (df_végétaux_dispo_intérieure['Disponibilité alimentaire protéines totale (Kg)']


Élément  Code zone     Zone  Code Produit Produit  Année Radical  Population  \
0                1  Arménie          2511     Blé   2013  plante     2977000   
1                1  Arménie          2513    Orge   2013  plante     2977000   
2                1  Arménie          2514    Maïs   2013  plante     2977000   
3                1  Arménie          2515  Seigle   2013  plante     2977000   
4                1  Arménie          2516  Avoine   2013  plante     2977000   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
0                         93.0                                    0.0   
1                        137.0                                   26.0   
2                         96.0                                    NaN   
3                          1.0                                    NaN   
4                          4.0                                    NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  Traitement  \
0                                                1024.0  ...        10.0   
1                                                   0.0  ...         7.0   
2                                                   0.0  ...         NaN   
3                                                   1.0  ...         NaN   
4                                                   2.0  ...         NaN   

Élément  Variation de stock  Disponibilité alimentaire protéines totale (Kg)  \
0                    -118.0                                      33163184.60   
1                       0.0                                             0.00   
2                       NaN                                         10866.05   
3                       0.0                                         21732.10   
4                       NaN                                         97794.45   

Élément  Disponibilité alimentaire totale (Kcal)  \
0                                   1.112684e+12   
1                                   0.000000e+00   
2                                   0.000000e+00   
3                                   1.086605e+09   
4                                   2.173210e+09   

Élément  Ratio énergie/poids (kcal/kg)  Pourcentage de protéines produit  \
0                          2860.368946                          8.525240   
1                                  NaN                               NaN   
2                                  NaN                               NaN   
3                                  NaN                               NaN   
4                          2173.210000                          9.779445   

Élément  Dispo intérieure (Kcal)  Dispo intérieure protéines  \
0                   1.584644e+12                4.722983e+07   
1                            NaN                         NaN   
2                            NaN                         NaN   
3                            NaN                         NaN   
4                   1.303926e+10                5.867667e+05   

Élément  Dispo végétaux_animaux_pertes(Kcal)  \
0                               1.470230e+12   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                               1.086605e+10   

Élément  Dispo végétaux_animaux_pertes protéines(Kg)  
0                                       4.381973e+07  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4                                       4.889722e+05  

[5 rows x 30 columns]

In [153]:
# Aggrégation par produit avec somme.
df_végétaux_dispo_intérieure_produits = df_végétaux_dispo_intérieure.groupby(by = 'Produit').sum().reset_index()

In [154]:
# On calcul les sommes au niveau mondiale
dispo_alimentaire_modiale_végétaux_kcal = df_végétaux_dispo_intérieure_produits['Dispo végétaux_animaux_pertes(Kcal)'].sum()
dispo_alimentaire_modiale_végétaux_protéines_kg = df_végétaux_dispo_intérieure_produits['Dispo végétaux_animaux_pertes protéines(Kg)'].sum()

In [155]:
dispo_alimentaire_modiale_végétaux_protéines_kg

160423201848.1706

In [156]:
dispo_alimentaire_modiale_végétaux_kcal



6103347225790079.0

In [157]:
print('Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux et produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en calories est',
      "{0:,}".format(dispo_alimentaire_modiale_végétaux_protéines_kg));
print( 'pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en  protéines est'
      ,dispo_alimentaire_modiale_végétaux_protéines_kg/nb_population * 100, '%')

Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux et produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en calories est 160,423,201,848.1706
pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en  protéines est 2292.635813283111 %


In [158]:
print('Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux et produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en calories est',
      "{0:,}".format(dispo_alimentaire_modiale_végétaux_kcal));
print( 'pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en  protéines est'
      ,dispo_alimentaire_modiale_végétaux_kcal/nb_population * 100, '%')

Le nombre de personnes  de la population mondiale qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux et produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en calories est 6,103,347,225,790,079.0
pourcentage de le  nombre de personnes de la population mondiale  qui pourraient être nourries  si toute la disponibilité intérieure mondiale de produits végétaux produits végétaux la nourriture végétale destinée aux animaux et les pertes de produits végétaux étaient utilisés pour de la nourriture en  protéines est 87223994.219936 %


# Question 9 :
Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? Donnez les résultats en termes de calories, puis de protéines, et exprimez ensuite ces 2 résultats en pourcentage de la population mondiale.

In [ ]:
df_vegetaux_animaux

In [161]:
# Sommes des disponibilités alimentaires  pour animaux et végétaux en Kcal
dispo_vegetaux_animaux_globale_kcal = df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)'].sum()
# Sommes des disponibilités alimentaires  pour animaux et végétaux en protéines
dispo_vegetaux_animaux_globale_protéines_kg = df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)'].sum()

In [162]:
print('Le nombre de personnes qui peuvent être nourries avec la disponibilité de nourriture dans le monde n en termes de calories est ', dispo_vegetaux_animaux_globale_kcal,'.')
print("Le nombre d'êtres humains qui peuvent être nourris avec la disponibilité de nourriture dans le monde n où se trouvent les protéines est ", dispo_vegetaux_animaux_globale_protéines_kg,'.')

Le nombre de personnes qui peuvent être nourries avec la disponibilité de nourriture dans le monde n en termes de calories est  7363883420185000.0 .
Le nombre d'êtres humains qui peuvent être nourris avec la disponibilité de nourriture dans le monde n où se trouvent les protéines est  207259180075.55 .
